### ISSUE TITLE
Make 'save' buttons float right 

### ISSUE THREAD

**Issue by rvanlaak** [WeblateOrg/weblate#1062]
Quick UX improvements (see #1057):
- Let all 'save' buttons float right? People are right action oriented.
##

**Comment by paulpach** [comment]
I don't like this one.

Currently the typical workflow is really nice:

You see the source string
You see the translation string
You see the "needs review" checkbox
You see the Save button.

Everything is left aligned and it reads intuitively from top to bottom.
If the save button floats right,  then you start going top to bottom and
all of a sudden the attention is supposed to shift far away off to the
right.  More so in wide screens.

Also,  if you put the save button off to the right,  by Fitts law
<https://en.wikipedia.org/wiki/Fitts%27s_law>,  you make it harder to click.

**Comment by rvanlaak** [comment]
Of course I agree with your law @paulpach :+1: this really is an "emotional" thing, given everything you can find about it:

http://www.creativebloq.com/netmag/why-people-focus-right-hand-side-design-3146267
http://uxmovement.com/buttons/why-ok-buttons-in-dialog-boxes-work-best-on-the-right/
http://ux.stackexchange.com/questions/8569/submit-buttons-on-the-left-or-the-right
http://ux.stackexchange.com/questions/37/should-the-ok-cancel-buttons-be-aligned-right-or-centered

Basically I'd expect buttons on the right if there are multiple actions (like the review checkbox). If you think of it as an application, then 'cancel / go back' should be on the left if you goto the previous page right? 

If you'd see the translation window as a more interactive / complex page in the near future, with more interactions in the bar with the save button, would you then expect the button at the right? Like here at the comment box of Github?

**Comment by paulpach** [comment]
That is very interesting reading,  tx.

Most web forms I have seen are left aligned and they usually keep the
submit button left aligned.  Especially forms that have high level of
linearity:  do a, b, c, then submit. Perhaps like you say, if the
interaction was more complex, and less linear...

Most dialog boxes I have seen have the ok button right aligned.  And your
reading material provides great insight as to why that is.

I don't know if there is a logical reason for web forms to be different
than dialog boxes, or if there has been some actual usability testing done
to determine which works best,  I can't seem to find any hard data.

At any rate,  keep in mind I am just a user giving an opinion,  I am not
the one you need to convince :).

Of course I agree with your law @paulpach <https://github.com/paulpach> [image:
:+1:] this really is an "emotional" thing, given everything you can find
about it:

http://www.creativebloq.com/netmag/why-people-focus-right-hand-side-design-3146267
http://uxmovement.com/buttons/why-ok-buttons-in-dialog-boxes-work-best-on-the-right/
http://ux.stackexchange.com/questions/8569/submit-buttons-on-the-left-or-the-right
http://ux.stackexchange.com/questions/37/should-the-ok-cancel-buttons-be-aligned-right-or-centered

Basically I'd expect buttons on the right if there are multiple actions
(like the review checkbox). If you think of it as an application, then
'cancel / go back' should be on the left if you goto the previous page
right?

If you'd see the translation window as a more interactive / complex page in
the near future, with more interactions in the bar with the save button,
would you then expect the button at the right? Like here at the comment box
of Github?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/nijel/weblate/issues/1062#issuecomment-207488296>

**Comment by nijel** [comment]
Currently Weblate has all buttons left aligned (unless you are using RTL translation :-)). I really do not see problem in changing that, but I'd really like to see bigger consensus on such change.

**Comment by nijel** [comment]
Revisiting this few years later, I don't see compelling reason to change this. I pretty much agree to what is written in https://ux.stackexchange.com/a/13539.
